source/extensions/common/wasm/wasm.cc

```diff
@@ -1194,7 +1194,11 @@ WasmResult Context::getProperty(absl::string_view path, std::string* result) {
         value = CelValue::CreateMessage(plugin_->listener_metadata_, &arena);
         break;
       case PropertyToken::CLUSTER_NAME:
-        value = CelValue::CreateString(&info->upstreamHost()->cluster().name());
+        if (!info->upstreamHost()->cluster().name().empty()) {
+          value = CelValue::CreateString(&info->upstreamHost()->cluster().name());
+        } else {
+          value = CelValue::CreateString(&info->routeEntry()->clusterName());
```

Comment:
I think routeEntry can be null? In any case, many of these pointers can be null, so we should harden it at some point.

Reply:
thanks! Done.